PLANNER-737: Workbench: scoreHolder global variable generation based on the PlanningSolution score type

mcimbora opened the pull request:
Based on https://github.com/droolsjbpm/kie-wb-common/pull/693 (not integrated yet), changes included in https://github.com/droolsjbpm/kie-wb-common/commit/0c6301c2332b070a94b2d09c962e3338fc937b6d

Add Save and Rename Helpers for Data Modeler. DataModelerRenameWorkaroundHelper is a temporary solution until we decide to switch to using RenameService when performing rename operation in DataModelerServiceImpl (currently there's a workaround which uses IOService.move).

Part of:
https://github.com/droolsjbpm/kie-wb-common/pull/699
https://github.com/droolsjbpm/optaplanner-wb/pull/136

manstis commented on kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceImpl.java:
```diff
@@ -758,6 +772,12 @@ public Path renameWorkaround( final Path path,
 
+                if ( renameHelperInstance != null ) {
+                    for ( DataModelerRenameWorkaroundHelper renameHelper : renameHelperInstance ) {
+                        renameHelper.postProcess( Paths.convert( _path ), Paths.convert( _target ) );
```
We can hope the Java compiler moves these conversions outside the loop....

manstis commented on kie-wb-common-screens/kie-wb-common-data-modeller/kie-wb-common-data-modeller-backend/src/main/java/org/kie/workbench/common/screens/datamodeller/backend/server/DataModelerServiceImpl.java:
```diff
@@ -570,7 +578,12 @@ public GenerationResult saveSource( final String source,
                 result.setPath( path );
+            }
 
+            if ( saveHelperInstance != null ) {
```
Now you have possible additional ```IOService``` operations; should the whole ```saveSource(...)``` method use a VFS "batch" (at the moment it's discretionary *only* if the package has changed).

mcimbora replied:
Correct, I checked IOService implementation we use and the concern of RenameService invoking startBatch once more is groundless. Invocations performed within the same thread share the same lock instance and the RenameService decreases the lock counter in a finally block.

mcimbora commented:
@manstis Updated, thanks for a review.

manstis commented:
@mcimbora You decided against adding unit tests?

mcimbora commented:
@manstis I just forget how I love writing unit tests :-) I added a couple of them for save/rename. Additionally I introduced better division of weld/mockito tests.

mcimbora commented:
@manstis Cool. I'll feel even better if the tests actually pass, looking.

mcimbora commented:
Jenkins retest this.

mcimbora commented:
@manstis All green, I think this is good to go in.
